### PR TITLE
Sensitive content filter

### PR DESCRIPTION
**Pull request by anton202**
## Fixes

Related to WordPress/openverse#636 by @panchovm 

## Description

This PR creates a Toggle switch and a new senesetive content filter
senesetive-content

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: feat/feature-flags.json [hunk 1]
```diff
@@ -28,6 +28,14 @@
       "status": "switchable",
       "description": "Can be switched between on and off",
       "defaultState": "on"
+    },
+    "toggle_senetive_content": {
```
Comment (obulat) [inline feat/feature-flags.json:32]:
```suggestion
    "toggle_sensitive_content": {
```

File: feat/feature-flags.json [hunk 2]
```diff
@@ -28,6 +28,14 @@
...
       "description": "Can be switched between on and off",
       "defaultState": "on"
+    },
+    "toggle_sensitive_content": {
+      "status": {
+        "staging": "enabled",
+        "production": "disabled"
+      },
+      "description": "Toggle sensitive content",
+      "defaultState": "off"
```
Comment (zackkrida) [inline feat/feature-flags.json:38]: Great!

File: src/components/VFilters/VSearchGridFilter.vue [hunk 3]
```diff
@@ -24,14 +24,22 @@
...
-      <VFilterChecklist
-        v-for="filterType in filterTypes"
-        :key="filterType"
-        :options="filters[filterType]"
-        :title="filterTypeTitle(filterType)"
-        :filter-type="filterType"
-        @toggle-filter="toggleFilter"
-      />
+      <template v-for="(filterType, index) in filterTypes">
+        <!-- seperation line betwen safe  browsing category and all the rest -->
```
Comment (zackkrida) [inline src/components/VFilters/VSearchGridFilter.vue:28]:
```suggestion
        <!-- Divider above the sensitive content filter -->
```

**Review by zackkrida — CHANGES_REQUESTED** [review]
This is phenomenal work! I love the new toggle component.

One thing we will need to fix, however, is accessibility. We need some focus styles for the toggle. Right now I can't tell if it can be switched using the keyboard or not.

@panchovm do you know if we have focus styles for this toggle yet? This may be blocked by that design work if not.

Thanks so much @anton202.

**Comment by anton202** [comment]
@zackkrida @olga thank you for your comments and changes and thx for correcting my english in each PR :)))
i`ll commit zack`s chaanges on sunday.
regarding accessibility, never really addreessed accessibility stuff before. so as you mentioned, i`ll add the ability to switch the toggle button using the keyboard.
regading the focus style, of course if @panchovm have an answear for it that will be great, but any way i`ll check in the design library that i took this switch toggle from (i think it was from wordpress design library)

**Comment by panchovm** [comment]
The toggle element used in Openverse is the one from the [WordPress Design Library](https://www.figma.com/file/e4tLacmlPuZV47l7901FEs/WordPress-Design-Library?node-id=28978%3A128675&t=DwnjYntGgUTWexK1-4). So please use that one and replace the focus color with openverse `pink`.

image

I also noticed that [this ticket](https://github.com/WordPress/gutenberg/issues/46734) addresses a component change. Not sure if it covers the toggle element but worth checking its scope. In any case, we need to follow the last version of WordPress' toggle component.

**Pull request closed without merging**